feature-libs/cart/saved-cart/components/details/saved-cart-details-overview/saved-cart-details-overview.component.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  Component,
+  ElementRef,
+  OnDestroy,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { Cart, TranslationService } from '@spartacus/core';
+import { Card, ICON_TYPE } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { filter, map, take } from 'rxjs/operators';
+import { SavedCartFormLaunchDialogService } from '../../saved-cart-form-dialog/saved-cart-form-launch-dialog.service';
+import { SavedCartDetailsService } from '../saved-cart-details.service';
+
+@Component({
+  selector: 'cx-saved-cart-details-overview',
+  templateUrl: './saved-cart-details-overview.component.html',
+})
+export class SavedCartDetailsOverviewComponent implements OnDestroy {
+  private subscription = new Subscription();
+
+  @ViewChild('element') element: ElementRef;
+
+  iconTypes = ICON_TYPE;
+  savedCart$: Observable<
+    Cart | undefined
+  > = this.savedCartDetailsService.getCartDetails();
+
+  constructor(
+    protected savedCartDetailsService: SavedCartDetailsService,
+    protected translation: TranslationService,
+    protected savedCartFormLaunchDialogService: SavedCartFormLaunchDialogService,
+    protected vcr: ViewContainerRef
+  ) {}
+
+  getCartName(cartName: string): Observable<Card> {
+    return this.translation.translate('savedCartDetails.cartName').pipe(
+      filter(() => Boolean(cartName)),
+      map((textTitle) => ({
+        title: textTitle,
+        text: [cartName],
+      }))
+    );
```

Comment:
Generally we should prefer using `cxTranslate` pipe in the template pipe over the `TranslationService` in the TS logic (unless `TranslationService` is justified). 

The disadvantages of using `TranslationService` are:
- it introduces an unnecessary dependency in the component TS class 
- it likely introduces unnecessary methods and RxJs logic to the TS class, which could be avoided

In this particular example we could have in the template:
```html
<cx-card [content]="{ 
   title: 'savedCartDetails.cartName' | cxTranslate
   text: [cart?.name],
}"></cx-card>
```